client/shared/src/main/scala/org/http4s/client/middleware/Metrics.scala

```diff
@@ -68,18 +72,24 @@ object Metrics {
     *
     * @param ops a algebra describing the metrics operations
     * @param classifierF a function that allows to add a classifier that can be customized per request
+    * @param tagsF a function that allows to add tags that can be customized per request
     * @param client the [[Client]] to gather metrics from
     * @return the metrics middleware wrapping the [[Client]]
     */
-  def effect[F[_]](ops: MetricsOps[F], classifierF: Request[F] => F[Option[String]])(
+  def effect[F[_]](
+      ops: MetricsOps[F],
+      classifierF: Request[F] => F[Option[String]],
+      tagsF: Request[F] => F[Map[String, String]],
```

Comment:
Hey! Thanks for the contribution. This kind of change for public API isn't binary compatible, and we are prohibited from doing such things. More information you can find in the [CI run](https://github.com/http4s/http4s/actions/runs/4004446458/jobs/6873576535#step:11:50).